### PR TITLE
chore: Iterative updates to NameSystem/DHT as validation integrated and exposed throughout.

### LINKED ISSUE #268: Remove configurable validation from NNS
As of #253 it is possible for us to inject validation logic from the entrypoint of the Noosphere Name System. However, it is quite possible that this is an anti-pattern (even though it makes building early tests very convenient). In the end, we want all implementations of the name system to rely on prescribed validation logic, so we probably don't want validation to be pluggable at all.

### PR DESCRIPTION
**Pull request by jsantell**
Iterative updates as #267 is being worked on in tests

* Propagate IPFS Client through Gateway workers.
* Use UcanJwtStore, rather than Custom Validator, for NameSystem configuration (fixes #268), though there is still currently an escape hatch for validation via None.
* Decouple sphere-esque mocks from NS bootstrapping networks in tests via `noosphere_ns::helpers::NameSystemNetwork`.
* Begin migration where DidParser is optional for `NsRecord::validate()`, and more specific errors during record validation.

### CODE REVIEW

**Review by cdata — CHANGES_REQUESTED** [review]
Nice cleanup, just a few small items to take care of :beers:

File: rust/noosphere-ns/Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,8 @@ readme = "README.md"
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
 anyhow = "^1"
-tracing = { workspace = true}
+tracing = { workspace = true }
+thiserror = "^1"
```
[the remark this answers is not in the document]
Comment (jsantell) [reply rust/noosphere-ns/Cargo.toml:29]: 100%

File: rust/noosphere-gateway/src/route/push.rs [hunk 2]
```diff
@@ -100,8 +100,8 @@ where
 
         // These steps are order-independent
         let _ = tokio::join!(
+            self.notify_ipfs_syndicator(next_version),
```
[the remark this answers is not in the document]
Comment (jsantell) [reply rust/noosphere-gateway/src/route/push.rs:103]: Nope, undoing

**Review by cdata — APPROVED** [review]
Woot!

**Pull request merged**